src/plugins/condition/components/Criterion.vue

```diff
@@ -149,52 +147,82 @@ export default {
             return type;
         }
     },
+    watch: {
+        telemetry: {
+            handler(newTelemetry, oldTelemetry) {
+                this.checkTelemetry();
+            },
+            deep: true
+        }
+    },
     mounted() {
         this.updateMetadataOptions();
     },
     methods: {
         checkTelemetry() {
-            if(this.criterion.telemetry &&
-                !this.telemetry.find((telemetryObj) => this.openmct.objects.areIdsEqual(this.criterion.telemetry, telemetryObj.identifier))) {
-                //telemetry being used was removed. So reset this criterion.
-                this.criterion.telemetry = '';
-                this.criterion.metadata = '';
-                this.criterion.input = [];
-                this.criterion.operation = '';
-                this.persist();
+            if(this.criterion.telemetry) {
+                if (this.criterion.telemetry === 'any' && this.criterion.telemetry === 'all') {
+                    this.updateMetadataOptions();
+                } else {
+                    if (!this.telemetry.find((telemetryObj) => this.openmct.objects.areIdsEqual(this.criterion.telemetry, telemetryObj.identifier))) {
+                        //telemetry being used was removed. So reset this criterion.
+                        this.criterion.telemetry = '';
+                        this.criterion.metadata = '';
+                        this.criterion.input = [];
+                        this.criterion.operation = '';
+                        this.persist();
+                    }
+                }
             }
         },
         getOperationFormat() {
             this.enumerations = [];
-            this.telemetryMetadata.valueMetadatas.forEach((value, index) => {
-                if (value.key === this.criterion.metadata) {
-                    let valueMetadata = this.telemetryMetadataOptions[index];
-                    if (valueMetadata.enumerations !== undefined) {
-                        this.operationFormat = 'enum';
-                        this.enumerations = valueMetadata.enumerations;
-                    } else if (valueMetadata.hints.hasOwnProperty('range')) {
-                        this.operationFormat = 'number';
-                    } else if (valueMetadata.hints.hasOwnProperty('domain')) {
-                        this.operationFormat = 'number';
-                    } else if (valueMetadata.key === 'name') {
-                        this.operationFormat = 'string';
-                    } else {
-                        this.operationFormat = 'string';
+            if (this.criterion.telemetry && (this.criterion.telemetry === 'any' || this.criterion.telemetry === 'all')) {
+                this.operationFormat = 'all';
```

Comment:
This seems semantically confusing to me. If value is 'any' it shouldn't be labeled as 'all'. 'total,' 'comprehensive,' 'integrated' would be possible labels encompassing both states.

Reply:
Agreed. wIll change it.